test/framework/src/main/java/org/elasticsearch/transport/nio/EventHandler.java

```diff
@@ -68,13 +69,23 @@ public void uncaughtException(Exception exception) {
      *
      * @param channel that should be closed
      */
-    public void handleClose(NioChannel channel) {
+    void handleClose(NioChannel channel) {
         channel.closeFromSelector();
         CloseFuture closeFuture = channel.getCloseFuture();
         assert closeFuture.isDone() : "Should always be done as we are on the selector thread";
         IOException closeException = closeFuture.getCloseException();
         if (closeException != null) {
-            logger.debug("exception while closing channel", closeException);
+            closeException(channel, closeException);
         }
     }
+
+    /**
+     * This method is called when an attempt to close a channel throws an exception.
+     *
+     * @param channel that was being closed
+     * @param exception that occurred
+     */
+    void closeException(NioChannel channel, Exception exception) {
+        logger.debug(new ParameterizedMessage("exception while closing channel: {}", channel), exception);
```

Comment:
Since this is at the debug level, should you use a supplier here to avoid instantiating the parameterized message if not needed?